Paintroid/src/main/java/org/catrobat/paintroid/MainActivity.java

```diff
@@ -296,14 +298,66 @@ protected void onDestroy() {
 		PaintroidApplication.savedPictureUri = null;
 		PaintroidApplication.saveCopy = false;
 
-		PaintroidApplication.commandManager.storeCommandLists();
 		PaintroidApplication.commandManager.setInitialized(false);
+		PaintroidApplication.commandManager.resetAndClear(false);
+
+		PaintroidApplication.colorPickerInitialColor = Color.BLACK;
 
 		IndeterminateProgressDialog.getInstance().dismiss();
 		ColorPickerDialog.getInstance().dismiss();
 		super.onDestroy();
 	}
 
+	@Override
+	public void onConfigurationChanged(Configuration newConfig) {
+		super.onConfigurationChanged(newConfig);
+
+		if (newConfig.orientation == 2)
```

Comment:
do not compare to hardcoded values, use Configuration.ORIENTATION_LANDSCAPE instead